src/NaCl.Core/XSalsa20.cs

```diff
@@ -23,7 +23,7 @@ public class XSalsa20 : Salsa20Base
         public XSalsa20(ReadOnlyMemory<byte> key, int initialCounter = 0) : base(key, initialCounter) { }
 
         /// <inheritdoc />
-        protected override void SetInitialState(Span<uint> state, ReadOnlySpan<byte> nonce, int counter)
+        internal protected override void SetInitialState(Span<uint> state, ReadOnlySpan<byte> nonce, int counter)

```

Comment:
Something minor... the access modifier should be `protected internal`, just need to fix the ordering.